plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -145,7 +148,14 @@ public void commit(@Nullable TableMetadata base, TableMetadata metadata)
         }
 
         if (base == null) {
-            commitNewTable(metadata);
+            if (PROVIDER_PROPERTY_VALUE.equals(metadata.properties().get(PROVIDER_PROPERTY_KEY))) {
+                // Assume this is a table executing migrate procedure
+                currentMetadataLocation = metadata.properties().get(METADATA_LOCATION_PROP);
+                commitToExistingTable(base, metadata);
```

Comment:
Iceberg table doesn't exist yet, so "commit to existing table" looks wrong.

this is similar problem to https://github.com/trinodb/trino/pull/14375#discussion_r985697010

Reply:
I chose `commitToExistingTable` because the table already exists in Hive metastore or Glue unlike #14375. Actually, I tried `commitNewTable` first, but it requires a change to call `HiveMetastore.replaceTable` & `AWSGlue#updateTable` in the method.